File: A's/cpp/2121A.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int t; cin >> t;

    while (t--) {
        int n, s; 
        cin >> n >> s;

        vector<int> values(n);
        
        for (int i = 0; i < n; i++) {
            cin >> values[i];
        }

        sort(values.begin(), values.end());
        int Lvalue = values.front();
        int Rvalue = values.back();

        // If s (starting pos) is already to the left (or right) of the whole interval we just need to walk it to the far end
        if (s <= Lvalue) {
            cout << Rvalue - s << endl;
        } else if (s >= Rvalue) {
            cout << s - Lvalue << endl;
        } else {
            // If s is in Lvalue, Rvalue, choose the cheaper "sweep" order
            int goL = 2 * (s - Lvalue) + (Rvalue - s);
            int goR = 2 * (Rvalue - s) + (s - Lvalue);
            cout << min(goL, goR) << endl; // Fastest "route"
        }
    }
    
    return 0;
}

File: A's/py/2121A.py
def main() -> None:
    t: int = int(input())

    while t:
        n, s = map(int, input().split())

        values: list[int] = list(map(int, input().split())).sort()

        Lval: int = values[0]
        Rval: int = values[-1]

        # If s (starting pos) is already to the left (or right) of the whole interval we just need to walk it to the far end
        if s <= Lval:
            print(Rval - s)
        elif s >= Rval:
            print(s - Lval)
        else:
            # If s is in Lvalue, Rvalue, choose the cheaper "sweep" order
            goL: int = 2 * (s - Lval) + (Rval - s)
            goR: int = 2 * (Rval - s) + (s - Lval)
            print(min(goL, goR)) # Fastest "route"

        t -= 1

if __name__ == "__main__":
    main()